cmd/jujud/agent/caasunitinit.go

```diff
@@ -95,6 +104,12 @@ func (c *CAASUnitInitCommand) Run(ctx *cmd.Context) (errOut error) {
 	// sending process. If neither wait or send is specified,
 	// continue with the arguments already specified.
 	if c.Wait {
+		defer func() {
+			// When a sender sends a call to us, wait for them to exit
+			// first, so when running under k8s, the container doesn't
+			// get killed before the sender has died.
+			c.waitForPIDFunc(c.Args.CallerPID)
```

Comment:
@wallyworld this should fix the issue you see with the 137 exit code. The container runtime was killing the container because the main process was exiting before the send process exited.